modules/message-lane/src/inbound_lane.rs

```diff
@@ -61,14 +61,24 @@ impl<S: InboundLaneStorage> InboundLane<S> {
 		}
 
 		data.latest_confirmed_nonce = outbound_lane_data.latest_received_nonce;
+		/// Firstly, remove all of the records where higher nonce <= new confirmed nonce
 		while data
 			.relayers
 			.front()
-			.map(|(nonce, _)| *nonce <= data.latest_confirmed_nonce)
+			.map(|(_, nonce_high, _)| *nonce_high <= data.latest_confirmed_nonce)
 			.unwrap_or(false)
 		{
 			data.relayers.pop_front();
 		}
+		/// Secondly, update all of the records with lower nonce equal to new confirmed nonce
+		for relayer in data.relayers.iter_mut() {
```

Comment:
Am I right that there must be at most 1 entry that falls under `*nonce_low <= data.latest_confirmed_nonce` condition? Basically you may only check `front()`

Reply:
You're right because the messages don't overlap for relayers I can just check the front. Updated in the latest commit.